From: Michal Kubecek <mkubecek@suse.cz>
Subject: kabi: handle addition of uevent_sock into struct net
Patch-mainline: Never, kabi workaround
References: bsc#1122982

Backport of mainline commit 94e5e3087a67 ("net: add uevent socket member")
adds uevent_sock member into struct net. Move this new member at the end of
the structure and hide it from genksyms. This is safe as struct net should
never be allocated directly or embedded in other structure or array.

Signed-off-by: Michal Kubecek <mkubecek@suse.cz>
---
 include/net/net_namespace.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/include/net/net_namespace.h
+++ b/include/net/net_namespace.h
@@ -78,8 +78,6 @@ struct net {
 	struct sock 		*rtnl;			/* rtnetlink socket */
 	struct sock		*genl_sock;
 
-	struct uevent_sock	*uevent_sock;		/* uevent socket */
-
 	struct list_head 	dev_base_head;
 	struct hlist_head 	*dev_name_head;
 	struct hlist_head	*dev_index_head;
@@ -156,6 +154,7 @@ struct net {
 		spinlock_t	lock;
 		u32		seq;
 	} ip6addrlbl_table;
+	struct uevent_sock	*uevent_sock;		/* uevent socket */
 	int			sysctl_tcp_min_snd_mss;
 	int			ip6frag_strict_short;
 #endif
